packages/actor-query-operation-group/lib/GroupsState.ts

```diff
@@ -86,11 +87,13 @@ export class GroupsState {
         await Promise.all(this.pattern.aggregates.map(async(aggregate) => {
           // If distinct, check first whether we have inserted these values already
           if (aggregate.distinct) {
```

Comment:
Thank you for this. Distinct handling in the expression evaluator does work as expected even though we don't supply it a variable to group by. That's because the GroupsState makes sure we use a seperate aggregator per aggregate.variable.

The new commit I'm pushing implements this solution, it solves the issue I referenced above.

Reply:
Amazing! Tnx!